mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -147,14 +150,63 @@ public RunInfo createRun(CreateRun request) {
    * @param runViewType ViewType for expected runs. One of (ACTIVE_ONLY, DELETED_ONLY, ALL)
    *                    Defaults to ACTIVE_ONLY.
    *
+   * @return A list of all RunInfos that satisfy search filter.
+   */
+  public List<RunInfo> searchRuns(List<String> experimentIds,
+                              String searchFilter,
+                              ViewType runViewType) {
+    SearchRuns.Builder builder = SearchRuns.newBuilder()
```

Comment:
does it make use a single `searchRuns` method to call REST endpoint and do the runs -> runInfo transformation here?

Reply:
If I'm understanding your question, it makes one call to the `searchRuns` REST endpoint that gives back Runs. Then it uses `stream().map()` to pick out all the RunInfo's out of the Runs. It doesn't call `searchRuns` or `listRunInfo` endpoints any more times.

Reply:
No. What I meant was could the body be reimplemented using `v2` methods.

```
return searchRuns(experimentIds, searchFilter, runViewType, new ArrayList<>(), new ArrayList<>(), null).getItems().stream().map(Run::getInfo).collect(Collectors.toList());
```